src/Microsoft.AspNet.Owin/OwinFeatureCollection.cs

```diff
@@ -427,6 +428,20 @@ public bool IsReadOnly
             get { return true; }
         }
 
+        Guid IRequestIdentifierFeature.TraceIdentifier
+        {
+            get
+            {
+                var requestId = Prop<string>(OwinConstants.RequestId);
+                if (requestId != null)
+                {
+                    return new Guid(requestId);
```

Comment:
Hmm this makes me more worried actually. If we want to support OWIN, and OWIN uses strings, I don't see how we can force GUIDs... that's not to say that if we happen to get a GUID that we shouldn't use it...